### ISSUE TITLE
Add an Option to Load Modules by ID in Header Blocks

### ISSUE THREAD

**Issue by salahjaafar** [templaza/astroid-framework#924]
#### Title: Add an Option to Load Modules by ID in Header Blocks  

Hello Astroid Framework Team,  

First of all, thank you for the incredible work you're doing with this framework, which makes Joomla! Development much easier.  

I’d like to propose an enhancement to add an option allowing us to load specific modules by their ID in **Header Block 1** and **Header Block 2**.  

### Proposal:  

The goal is to introduce functionality similar to Joomla’s existing `loadmoduleid` method (used in content). While this method simplifies loading modules by ID in content areas, it’s currently unavailable for the **Header Blocks**.  

**Challenges:**  
- Modules assigned to positions like `left`, `right`, etc., can sometimes display duplicate content when loaded in multiple contexts.  
- Assigning modules to specific positions isn’t always intuitive for dynamic configurations.  

### Benefits of This Feature:  

1. **Increased Flexibility:**  
   Loading modules directly by their ID would remove the need to assign them to specific positions.  

2. **Elimination of Duplicates:**  
   This approach would prevent duplicate displays, especially when modules are assigned to multiple positions.  

3. **Consistency with Joomla:**  
   Aligning this feature with Joomla's existing practices would improve integration and familiarity for users.  

### Example of Usage:  

Here’s a hypothetical implementation:  

```php
<astroid:loadmoduleid moduleid="123" block="header1" />
<astroid:loadmoduleid moduleid="456" block="header2" />
```  

Where:  
- `moduleid` specifies the module’s ID to be loaded.  
- `block` indicates the target area (e.g., `header1` or `header2`).  

### Conclusion:  

This enhancement would be a significant improvement for Astroid Framework users, especially for complex projects requiring dynamic modular displays without redundancy.  

Thank you for considering this suggestion. If you need additional details, I’d be happy to collaborate further to refine this proposal.  

Best regards,  
Salah JAAFAR

**Comment by felixkat** [comment]
HI @salahjaafar,

I was reading through your proposal but I'm not 100% sure if I am following what you are suggesting.  Does the following not meet your requirements?

![Image](https://github.com/user-attachments/assets/330b407e-cdd0-40b7-b647-5c3d0536da91)


Which allows you to put a module into a block.

**Comment by salahjaafar** [comment]
@felixkat 
In certain cases, you need to select the module ID and avoid assigning a position. This prevents the module from appearing in multiple positions.

**Comment by felixkat** [comment]
@salahjaafar If you want to test a solution for this one you can do it with an override, although I don't see any reason why we can't put it in the main code.

You'll need to make a copy of /libraries/astroid/framework/frontend/header/horizontal.php    (or sidebar, stacked, sticky, depending on what Header Mode you are using).

Just copy the file to ROOT/templates/YOURTEMPLATE/html/frontend/header/ folder to create an override

Change  (line 73) - `echo $block_2_custom;` to` echo JHtml::_('content.prepare', $block_2_custom);`

Change (line 130) - `echo $block_1_custom` to echo `JHtml::_('content.prepare', $block_1_custom);`


Then change your block to Custom HTML and do your usual {loadmoduleid moduleId}

![Image](https://github.com/user-attachments/assets/0152d2ba-6479-4dab-af73-9e8458c7f518)


Let me know if this works.

**Comment by salahjaafar** [comment]
Yes, it works perfectly. @sonvnn , please edit the file libraries/astroid/framework/frontend/header/horizontal.php and add the code mentioned by @felixkat . It provides more flexibility.

**Comment by felixkat** [comment]
@sonvnn Any objections implementing this as above?

I don't think it would add much processing but we could also add an extra option to "process content plugin", so it doesn't get run by default with Custom HTML.

**Comment by sonvnn** [comment]
@felixkat can you create a pull request for it?
Thanks, Sonny

**Comment by felixkat** [comment]
Yes I'll do it when I get a chance.  I'll add the option into all the configurations.

**Comment by felixkat** [comment]
https://github.com/templaza/astroid-framework/pull/1057

**Comment by felixkat** [comment]
I've added the option to "Process Joomla Content Plugin for Custom HTML" on all Custom HTML for all menu configurations.  

The option is to save a little processing time for those who use Custom HTML but don't need to process Content Plugins.


![Image](https://github.com/user-attachments/assets/2bebc466-977b-4bb6-a3ea-d77754b0dbea)

**Comment by salahjaafar** [comment]
@felixkat @sonvnn 

# Ensuring Compatibility with Joomla 5 and Joomla 6  

To make your code compatible with both Joomla 5 and Joomla 6, replace:  

```php
echo JHtml::_('content.prepare', $block_2_custom);
```  

with `HTMLHelper`. Here's how to proceed:  

### 1. Import the Class  
At the top of your PHP file, add the following line to import the `HTMLHelper` class:  

```php
use Joomla\CMS\HTML\HTMLHelper;
```  

### 2. Update Your Code  
Replace the existing line with:  

```php
echo HTMLHelper::_('content.prepare', $block_2_custom);
```  

### Alternative: Using the Full Namespace  
If you cannot use the `use` statement (e.g., in some template files), use the full class name instead:  

```php
echo \Joomla\CMS\HTML\HTMLHelper::_('content.prepare', $block_2_custom);
```  

### Why Use `HTMLHelper`?  
This ensures your code works in Joomla 5, where `JHtml` is still supported via a compatibility plugin, and in Joomla 6, where `JHtml` may be removed. Using `HTMLHelper` is the recommended approach to future-proof your code and prevent issues during upgrades.
